google/cloud/bigquery/client.py

```diff
@@ -2647,9 +2648,17 @@ def load_table_from_dataframe(
                         job_config.schema,
                         tmppath,
                         parquet_compression=parquet_compression,
+                        parquet_use_compliant_nested_type=True,
                     )
                 else:
-                    dataframe.to_parquet(tmppath, compression=parquet_compression)
+                    dataframe.to_parquet(
+                        tmppath,
+                        engine="pyarrow",
+                        compression=parquet_compression,
+                        **{"use_compliant_nested_type": True}
+                        if _helpers.PYARROW_VERSIONS.use_compliant_nested_type
+                        else {},
```

Comment:
Let's wrap this to be clear that we are doing the `**` on the `else` branch, too.

```suggestion
                        **(
                            {"use_compliant_nested_type": True}
                            if _helpers.PYARROW_VERSIONS.use_compliant_nested_type
                            else {}
                        ),
```